v2/internal/controllers/samples_test.go

```diff
@@ -106,23 +110,58 @@ func runGroupTest(tc *testcommon.KubePerTestContext, groupVersionPath string) {
 	tc.DeleteResourceAndWait(rg)
 }
 
-func processSamples(tc *testcommon.KubePerTestContext, samples map[string]genruntime.ARMMetaObject) []client.Object {
+func processSamples(samples map[string]genruntime.ARMMetaObject) []client.Object {
 	samplesSlice := make([]client.Object, 0, len(samples))
 
 	for _, resourceObj := range samples {
 		obj := resourceObj.(client.Object)
-
-		findRefsAndCreateSecrets(tc, resourceObj)
 		samplesSlice = append(samplesSlice, obj)
 	}
 
 	return samplesSlice
 }
 
-func findRefsAndCreateSecrets(tc *testcommon.KubePerTestContext, resourceObj genruntime.ARMMetaObject) {
-	refs, err := reflecthelpers.FindSecretReferences(resourceObj)
-	tc.Expect(err).To(BeNil())
-	for ref := range refs {
+// findRefsAndCreateSecrets finds all references not matched by a corresponding genruntime.SecretDestination and generates
+// secrets which correspond to those references
+func findRefsAndCreateSecrets(tc *testcommon.KubePerTestContext, samples map[string]genruntime.ARMMetaObject, refs map[string]genruntime.ARMMetaObject) {
+	allDestinations := set.Make[genruntime.SecretDestination]()
+	allReferences := set.Make[genruntime.SecretReference]()
+
+	// Join samples and refs
+	resources := make(map[string]genruntime.ARMMetaObject, len(samples)+len(refs))
+	for k, v := range samples {
+		resources[k] = v
+	}
+	for k, v := range refs {
+		resources[k] = v
+	}
```

Comment:
Is there any possibility that `samples` and `refs` will have keys in common? 
If there _is_, the current handling is that `refs` wins by overwrite - is that correct?

Even if the current behaviour is correct, we need a comment explaining that.

Reply:
There's not really any possibility that they collide because the key is filepath. With that said I added:
```
		// It shouldn't be possible for these to collide as the key is a file path and samples and refs should have distinct
		// file paths given they're in different folders. In case somehow this does happen, we panic
		if _, ok := resources[k]; ok {
			panic(fmt.Sprintf("unexpected resource sample and ref collision at %s", k))
		}
```

Reply:
Doh, they _can_ collide. Fixed this to prevent that possibility.